src/audioWorklet/recorderProcessor.js

```diff
@@ -30,22 +33,26 @@ class RecorderProcessor extends AudioWorkletProcessor {
     }
 
     const input = inputs[0];
-    const output = outputs[0];
-
-    for (let channel = 0; channel < output.length; ++channel) {
-      const inputChannel = input[channel];
-      if (channel === 0) {
-        this.leftBuffers.push(inputChannel);
-        if (this.numInputChannels === 1) {
-          this.rightBuffers.push(inputChannel);
+
+    this.inputRingBuffer.push(input);
+
+    if (this.inputRingBuffer.framesAvailable >= this.bufferSize) {
```

Comment:
the switch to Ring Buffer is causing an issue in Firefox for "/examples/record". It seems that the bufferSize for Firefox ScriptProcessorNode is 4096, regardless of what we pass in here.